src/olgm/gm.js

```diff
@@ -93,26 +95,31 @@ olgm.gm.createGeometry = function(geometry, opt_ol3map) {
   var inProj = (opt_ol3map !== undefined) ?
     opt_ol3map.getView().getProjection() : 'EPSG:3857';
 
-  var latLngs = [];
-  var lonLatCoords;
+  var genLatLngs = function(coordinates) {
+    var latLngs = [];
+    var lonLatCoords;
 
-  var coordinates;
-  if (geometry instanceof ol.geom.LineString) {
-    coordinates = geometry.getCoordinates();
-  } else {
-    coordinates = geometry.getCoordinates()[0];
-  }
-
-  for (var i = 0, len = coordinates.length; i < len; i++) {
-    lonLatCoords = ol.proj.transform(coordinates[i], inProj, 'EPSG:4326');
-    latLngs.push(new google.maps.LatLng(lonLatCoords[1], lonLatCoords[0]));
-  }
+    for (var i = 0, len = coordinates.length; i < len; i++) {
+      lonLatCoords = ol.proj.transform(coordinates[i], inProj, 'EPSG:4326');
+      latLngs.push(new google.maps.LatLng(lonLatCoords[1], lonLatCoords[0]));
+    }
+    return latLngs;
+  };
 
   var gmapGeometry = null;
-  if (geometry instanceof ol.geom.LineString) {
-    gmapGeometry = new google.maps.Data.LineString(latLngs);
+
+  if (geometry instanceof ol.geom.MultiPolygon) {
```

Comment:
Let's move this at the end, i.e.:

1: if line string
2: else if polygons
3: else (multi polygons)